### PR TITLE
Adds section to Provider spec regarding optional createKey function

### PR DESCRIPTION
**Pull request by rgwozdz**

### CODE REVIEW

File: docs/specs/provider.md [hunk 1]
```diff
@@ -99,6 +99,13 @@ The position of the provider-specific fragment of a route path can vary dependin
 ### Output-routes without provider parameters
 You may need routes that skip the addition of provider-specific parameters altogether.  This can be accomplished by adding an `absolutePath: true` key-value to the `routes` array object in your output-services plugin. On such routes, Koop will define the route without any additional provider namespace or parameters.
 
+### Function: createKey
```
Comment (dmfenton) [inline docs/specs/provider.md]: do we have a table anywhere of the required/optional functions on the provider? I think we need one at this point:
getData: required => docs
authorize: optional => docs
authenticate: optional => docs
createKey: optional => docs

Comment (rgwozdz) [reply docs/specs/provider.md]: I agree a table would be nice.  One complication with the above list is that `authorize` and `authenticate` are actually defined on the [authorization plugin](https://github.com/koopjs/koop-auth-direct-file/blob/master/src/index.js), and only added to each provider's prototype at runtime.  Should they still be included on the provider spec table?  I guess we could link to the authorization spec where they are described

Comment (dmfenton) [reply docs/specs/provider.md]: If you use an existing plugin. But a provider could still roll their own right?

Comment (rgwozdz) [reply docs/specs/provider.md]: That is true.  I hadn't thought of a developer doing it that way, but they certainly could.

**Pull request merged**